services/crosschainconnector/ethereum/service.go

```diff
@@ -155,19 +171,78 @@ func (s *service) EthereumGetTransactionLogs(ctx context.Context, input *service
 }
 
 func (s *service) EthereumGetBlockNumber(ctx context.Context, input *services.EthereumGetBlockNumberInput) (*services.EthereumGetBlockNumberOutput, error) {
-	logger := s.logger.WithTags(trace.LogFieldFrom(ctx))
-	logger.Info("getting current safe Ethereum block number")
+	//	logger := s.logger.WithTags(trace.LogFieldFrom(ctx))
+	//	logger.Info("getting current safe Ethereum block number")
 
-	ethereumBlockNumber, err := getFinalitySafeBlockNumber(ctx, input.ReferenceTimestamp, s.timestampFinder, s.config)
+	blockNumberAndTime, err := s.getFinalitySafeBlockNumber(ctx, input.ReferenceTimestamp)
 	if err != nil {
 		return nil, err
 	}
 
-	if ethereumBlockNumber == nil {
-		return nil, errors.Errorf("failed getting an actual current block number from Ethereum") // note: the geth simulator does not support this API
+	// TODO	https://github.com/orbs-network/orbs-network-go/issues/1214 simulator returns nil from FindBlockByTimestamp
+	if blockNumberAndTime == nil {
+		return nil, errors.Errorf("failed getting an actual current block number from Ethereum")
 	}
 
 	return &services.EthereumGetBlockNumberOutput{
-		EthereumBlockNumber: ethereumBlockNumber.Uint64(),
+		EthereumBlockNumber: uint64(blockNumberAndTime.BlockNumber),
+	}, nil
+}
+
+func (s *service) EthereumGetBlockNumberByTime(ctx context.Context, input *services.EthereumGetBlockNumberByTimeInput) (*services.EthereumGetBlockNumberByTimeOutput, error) {
+	blockNumberAndTime, err := s.getFinalitySafeBlockNumber(ctx, input.EthereumTimestamp)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO	https://github.com/orbs-network/orbs-network-go/issues/1214 simulator returns nil from FindBlockByTimestamp
+	if blockNumberAndTime == nil {
+		return nil, errors.Errorf("failed getting an actual current block number from Ethereum")
+	}
+
+	err = s.verifyBlockNumberIsFinalitySafe(ctx, uint64(blockNumberAndTime.BlockNumber), input.ReferenceTimestamp)
+	if err != nil {
+		return nil, err
+	}
+
+	return &services.EthereumGetBlockNumberByTimeOutput{
+		EthereumBlockNumber: uint64(blockNumberAndTime.BlockNumber),
+	}, nil
+}
+
+func (s *service) EthereumGetBlockTime(ctx context.Context, input *services.EthereumGetBlockTimeInput) (*services.EthereumGetBlockTimeOutput, error) {
```

Comment:
is this the latest block time? latest final? let's change the naming with @OdedWx so that it's crystal clear what each variant returns

Reply:
its clear.
if EthereumGetBlockNumber is : the newest block under finality then 
this means it's the time of the finality block

the only 100% clear name would be EthereumGetLatestBlockNumberUnderFinalit
and ... Time .... these names are too much.